File: CharlesAndKeith/app/src/main/java/com/zawzaw/candkassignment/viewpods/EmptyViewPod.java
package com.zawzaw.candkassignment.viewpods;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;

import com.bumptech.glide.request.RequestOptions;
import com.zawzaw.candkassignment.R;
import com.zawzaw.candkassignment.utils.GlideApp;

public class EmptyViewPod extends RelativeLayout {

    @BindView(R.id.iv_empty)
    ImageView ivEmpty;

    @BindView(R.id.tv_empty)
    TextView tvEmpty;

    public EmptyViewPod(Context context) {
        super(context);
    }

    public EmptyViewPod(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public EmptyViewPod(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        ButterKnife.bind(this, this);

    }

    public void setEmptyData(String imageUrl, String emptyMsg) {

        GlideApp.with(getContext())
                .load(imageUrl)
                .apply(RequestOptions.circleCropTransform())
                .into(ivEmpty);

        tvEmpty.setText(emptyMsg);

    }

    public void setEmptyData(int emptyImageResource, String emptyMsg) {

        ivEmpty.setImageResource(emptyImageResource);
        tvEmpty.setText(emptyMsg);
    }

}

File: MM-Healthcare/app/src/main/java/com/zawzaw/padc/mmhealthcare/viewpods/EmptyViewPod.kt
package com.zawzaw.padc.mmhealthcare.viewpods

import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import com.zawzaw.padc.mmhealthcare.R
import kotlinx.android.synthetic.main.view_pod_empty.view.*

/**
 * Created by zawzaw on 21/07/2018.
 */

class EmptyViewPod : RelativeLayout {

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, styleAttr: Int) : super(context, attrs, styleAttr)

    override fun onFinishInflate() {
        super.onFinishInflate()
    }

    fun setEmptyData(errorMessage: String, errorImage: Int) {
        tv_empty_text.text = errorMessage
        iv_empty_image.setImageResource(R.drawable.empty_data_placeholder)
    }

}
